### DISCUSSION TITLE
Spreading configuration over multiple files

### DISCUSSION THREAD

**Discussion by pkleineb** [zellij-org/zellij#3376] [Q&A]
Hi I am really new to zellij but it sounds great and looks really interesting.
I myself am a fan of how nvim uses lua for it's configs because the scripting aside you can spread the config into multiple files and then just require the files.
Some people might not like that approach but I think it makes your config really clean and tidy.

So I was wondering, if there is also a way for zellij to have it's configuration spread out over multiple files?
Since zellij is using kdl for configuration(which can't include files by how I understood it) is there maybe an option in zellij to use multiple config files just like one could do the same thing with layouts and themes by setting theme_dir and layouts_dir?

**Comment by cristiand391** [comment]
> Since zellij is using kdl for configuration(which can't include files by how I understood it) is there maybe an option in zellij to use multiple config files just like one could do the same thing with layouts and themes by setting theme_dir and layouts_dir?

There's no config like that yet.

You can always have different config files and load them when starting zellij like this:
https://zellij.dev/documentation/configuration#where-does-zellij-look-for-the-config-file

but it's not that they loaded config will be merged with the default `config.kdl`.

**Reply by pkleineb** [reply]
Mhm okay can one add multiple config files to zellij on the cli? And will that overwrite zellij default config?

**Reply by Lockszmith-GH** [reply]
a possible option would be to use a templating tool, like chezmoi, to create the kdl configuration in a more flexible manner.

**Reply by cristiand391** [reply]
> Mhm okay can one add multiple config files to zellij on the cli? And will that overwrite zellij default config?

@pkleineb looks like passing `--config-dir` doesn't fail but I don't see those being merged.

**Reply by pkleineb** [reply]
@Lockszmith-GH chezmoi doesn't seem to fit my exact use case i'd like some kind of import of sorts like you would for example in lua do:
```lua
require("keybinds")
require("themes")
```

**Reply by pkleineb** [reply]
@cristiand391 that is unlucky maybe the zellij devs will add including syntax as a later feature

**Comment by shaleh** [comment]
Are the Zellij maintainers opposed to supporting some form of "include" for config files? My use case is sharing Zellij across several machines. I want the structure of my layout but the actual commands need to change for work or personal or whatever. I would love to have one config, a common layout template file which defines a tab and pane template, and then the actual layout being used which imports / includes those templates and does what I need for the particular scenario.

**Reply by pkleineb** [reply]
Include would be a great idea for my use case as well but the problem may be with kdl it self since it doesn't support that syntax.

**Comment by shaleh** [comment]
What if we side step that, at least initially. What if we put a template in one file say default.templatr.kdl and then if that exists those templates are used when the layout is loaded.

Separately, I don't see why kdl needs to support this. XML and similar formats allowed the user/developer to define their own functionality. If an "include" is encountered it is up to the app parsing it to handle it.

**Reply by pkleineb** [reply]
fair enough i think that would be a nice feature

**Comment by ser** [comment]
i managed to overcome the issue on nixos, where I simply merge sets with layout, using https://home-manager-options.extranix.com/?query=programs.zellij.layouts&release=master
